packages/opentelemetry-core/src/common/attributes.ts

```diff
@@ -47,6 +47,47 @@ export function isAttributeValue(val: unknown): val is AttributeValue {
   return isValidPrimitiveAttributeValue(val);
 }
 
+export function truncateValueIfTooLong(
+  value: AttributeValue,
+  limit: number | null,
+  truncationWarningCallback = () => {}
+): AttributeValue {
+  if (limit === null) {
+    return value;
+  }
+
+  if (limit < 32) {
+    throw new Error('Value size limit cannot be lower than 32.');
```

Comment:
It should never throw any error, it can warn user but never throw.
with regards to limit I posted a reply on spec issue